Cleanup structure

skjnldsv opened the pull request:
Fix #4047

@nextcloud/designers 

This is a much cleaner structure for our software

## Done
- Removed `div#content-wrapper`
- Removed any absolute positionning of our content
  - Now the whole page is scrolled, not a div within an absolute div
  - Android header is now collapsing because the scrolling behaviour has been restored to its original state
- Merged `header` and `div#header` to follow html5 guidelines

## Deprecated and guidelines
- Do not use `#content-wrapper` anymore
- If your app is injecting itself by replacing the `#content` element, make sure to keep the `#content` id
- If you use the `app-content-list` standard, make sure to hide the `app-content-details` div when the list is visible in mobile mode (full screen) and hide the `app-content-list` when the details is visible
- This is  `app-content-details` and NOT `app-content-detail`


## Overall structure
``` html
<header>
    <div class="header-left">
        <!-- apps menu -->
    </div>
    <div class="header-right">
        <!-- search - contactsmenu - settingsmenu - ... -->
    </div>
</header>
<div id="content" class="app-YOURAPPID">
    <div id="app-navigation" class="">
        <div class="app-navigation-new">
            <!-- app 'new' button -->
        </div>
        <ul id="usergrouplist">
            <!-- app navigation -->
        </ul>
        <div id="app-settings">
            <!-- app settings -->
        </div>
    </div>
    <div id="app-content">
        <div id="app-navigation-toggle" class="icon-menu"></div>
        <!-- app-content-wrapper is optional, only use if app-content-list  -->
        <div id="app-content-wrapper">
            <div class="app-content-list">
                <!-- app list -->
            </div>
            <div class="app-content-details"></div>
            <!-- app content -->
        </div>
    </div>
    <div id="app-sidebar"></div>
</div>
```

skjnldsv committed:
Cleanup structure

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

stefan-niedermann commented:
May i ask why no `nav`-tag is used for `#app-navigation` or `.header-left`?

skjnldsv commented:
@stefan-niedermann We use a nav now in the app list https://github.com/nextcloud/server/blob/625bdf0e6c6e3ebcde102f22cf54cbd505d6faa8/core/templates/layout.user.php#L73

But not on the app navigation, which we should, you're completely right :)

rullzer commented:
conflicts and failing tests

juliushaertl commented:
While I like the new structure in general, scrolling the page instead of the app content causes a weired double scrollbar, when the sidebar is visible:
![image](https://user-images.githubusercontent.com/3404133/41899484-da1e3e0c-792c-11e8-90db-14f640fc2787.png)

skjnldsv commented:
@juliushaertl Ah damn, it make sense.
The sidebar should be fixed though, I'll change this so the scrollbar is on the real content then!

skjnldsv committed:
Adjust acceptance tests to sidebar changes in Files app

Before, each section of the Files app ("All files", "Favorites"...) had
its own sidebar element. Now there is a single sidebar element for all
the sections in the Files app.

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Sidebar fix
Fix sidebar and scrollbar on content
Sidebar fixes, and sidebar-width + header-height variables

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

MorrisJobke commented:
* [ ] the sidebar moves below the content on smaller screens:
![bildschirmfoto 2018-06-28 um 16 52 03](https://user-images.githubusercontent.com/245432/42042102-a837590a-7af3-11e8-97b3-9e28c6ab8469.png)

skjnldsv commented:
@juliushaertl hum, strange, I don't have this. What browser and did you cleaned the cache?

juliushaertl commented:
Ah ok, seems that was a caching issue. But https://github.com/nextcloud/server/pull/9982#issuecomment-401062597 still exists and the breadcrumbs are not sticky anymore in the files app.

Besides that I really like it. It's a lot cleaner than before :+1:

skjnldsv commented:
@juliushaertl yep, I have this to fix!
Incoming! :ship:

skjnldsv commented:
- [ ] Control sticky not working

@juliushaertl sticky is behaving strangely, do you know more about it than I do? :)

skjnldsv committed:
Fixed sticky controls

since controls are not directly into the scollable element (app-content)
we need to make the parent the scrollable element (app-content-files)

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@danxuliu could you help me on the acceptance tests? :)

skjnldsv committed:
use flex for main content

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fix sidebar close button

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Okay, last cleanup!!

We dropped position fixed on our main content and went to flex.
This way we have a proper flow for the navigation-content-sidebar.
Showing or hiding one or the other display a nice animation and is not made by strange hacks of negative margins and absolute/fixed positionning. :tada: 

@jancborchardt just realized most of the sidebar design is located in the files app...
We need to take care of this next week ;)

danxuliu commented:
I have adjusted the acceptance tests to the sidebar changes.

Note, however, that https://github.com/nextcloud/server/blob/master/tests/acceptance/features/app-files.feature#L14 still fails because it is actually a regression from the old code; just perform the steps manually to check the issue ;-)

jancborchardt reviewed (dismissed):
1. Reduce the width of the window
2. Open the right sidebar of a file, like for sharing
3. Open the left navigation using the menu icon
4. Both left and right elements overlap

jancborchardt commented:
On smaller widths, this happens. We should have a min-width or so for the names, or overlap the right sidebar instead of moving the content like we do now already:
![screenshot from 2018-07-17 11-17-16](https://user-images.githubusercontent.com/925062/42808236-17609d00-89b3-11e8-9240-940d8a6cc9d6.png)

jancborchardt committed:
Widen left navigation to 300px and set max-width of right sidebar to 500px

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Fixed filestable min-width

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Jsunit fixes 1

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Bump karma config

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

nickvergessen commented on core/templates/layout.public.php:
```diff
@@ -70,9 +69,7 @@
 		<?php } ?>
-	</div>
-</header>
-<div id="content-wrapper">
```
For the record, this will break mail, richdocuments and gallery

nickvergessen replied:
@morrisjobke did you see this?

skjnldsv replied:
@nickvergessen yes we saw. We'll adapt the apps to the proper layout

MorrisJobke replied:
@nickvergessen The point is: we can't keep the HTML DOM always stable, because at some point you need to change it :/ So the plan would be to have this in the beta and fix those apps then.

nickvergessen replied:
Just wanted to be sure that its not forgotten

skjnldsv committed:
Jsunit fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

danxuliu commented:
The issue mentioned in https://github.com/nextcloud/server/pull/9982#issuecomment-404581816 is still valid. My bad, I should have used a _Request changes_ comment instead of a regular comment ;-)

I have also noticed that this pull request causes the confirm icon in the share search to have a wrong position:
![share-search-icon](https://user-images.githubusercontent.com/26858233/43037639-916512a2-8d0f-11e8-87cb-dee6011d39c6.png)

And just for your information, besides the apps already mentioned, the layout of Talk and the public share page are also broken now.

skjnldsv commented:
@danxuliu Yep, this is my bad on this! 
I thought I checked the latest tests, I guess not!

This is fixed in #10218 though!
